model/flow/protocol_state.go

```diff
@@ -95,63 +138,52 @@ func NewRichProtocolStateEntry(
 				protocolState.PreviousEpochEventIDs.CommitID)
 		}
 	}
-	if protocolState.CurrentEpochEventIDs.SetupID != currentEpochSetup.ID() {
+	if protocolState.CurrentEpoch.SetupID != currentEpochSetup.ID() {
 		return nil, fmt.Errorf("supplied current epoch setup (%x) does not match protocol state (%x)",
 			currentEpochSetup.ID(),
-			protocolState.CurrentEpochEventIDs.SetupID)
+			protocolState.CurrentEpoch.SetupID)
 	}
-	if protocolState.CurrentEpochEventIDs.CommitID != currentEpochCommit.ID() {
+	if protocolState.CurrentEpoch.CommitID != currentEpochCommit.ID() {
 		return nil, fmt.Errorf("supplied current epoch commit (%x) does not match protocol state (%x)",
 			currentEpochCommit.ID(),
-			protocolState.CurrentEpochEventIDs.CommitID)
+			protocolState.CurrentEpoch.CommitID)
 	}
 
 	var err error
-	nextEpochProtocolState := protocolState.NextEpochProtocolState
+	nextEpoch := protocolState.NextEpoch
 	// if next epoch has been already committed, fill in data for it as well.
-	if nextEpochProtocolState != nil {
+	if nextEpoch != nil {
 		// sanity check consistency of input data
-		if nextEpochProtocolState.CurrentEpochEventIDs.SetupID != nextEpochSetup.ID() {
+		if nextEpoch.SetupID != nextEpochSetup.ID() {
```

Comment:
if we define that `ID()` method of `nil` instances of `EpochSetup` have  `ZeroID`, we might want to include a check here that `nextEpoch.SetupID != ZeroID`